sphinxcontrib/chapeldomain/__init__.py

```diff
@@ -56,14 +56,62 @@
            (?:[+*/!~%<>=&^|\-:]+)                  #  or operator name
           )  \s*
           (?:\((.*?)\))?                           # opt: arguments
-          (\s+(?:const\s)? (?:\w+?)|               #  or return intent
-           \s* : \s* (?:[^:]+?)|                   #  or return type
-           \s+(?:const\s)? \w+\s* : \s* (?:[^:]+?) #  or return intent and type
+          (\s+(?:const\s)? (?:\w+?)                # opt: return intent
+          )?
+          (\s*:\s*(?:[^:]+?)                       # opt: return type
+          )?
+          (\s+throws                               # opt: throws
           )?
           (\s+where\s+.*                           # Where clause
           )?
           $""", re.VERBOSE)
 
+
+def match_chpl_sig_pattern(sig: str):
+    """
+    Match a Chapel signature against the regex pattern defined in
+    chpl_sig_pattern. chpl_sig_pattern cannot be used directly because we need
+    to fix some things up
+    """
+    sig_match = chpl_sig_pattern.match(sig)
+    if not sig_match:
+        return None
+    (
+        func_prefix,
+        name_prefix,
+        name,
+        arglist,
+        return_intent,
+        return_type,
+        throws,
+        where_clause,
+    ) = sig_match.groups()
+
+    if return_type:
+        return_type = return_type.strip().removeprefix(':').lstrip()
+    if return_intent:
+        return_intent = return_intent.strip()
+    if throws:
+        throws = throws.strip()
+
+    if return_intent and "throws" in return_intent:
```

Comment:
A comment here for why this was necessary will help future developers :)

Reply:
I added this to the docstring